### DISCUSSION TITLE
About top level/class level Capsules

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#159] [General]
In docs you say that we must use top level or static capsules, but at the same time you're using a class level Capsule on `RearchConsumer.build`. So the question is if it is okay to use class level caspules if you know what are you doing?

**Comment by GregoryConrad** [comment]
Can you point to this "class level Capsule?" I'm not sure exactly what you're referring to.

You can use "inline capsules" within a capsule or widget build safely (without causing leaks), but normally all other capsules must be top-level.

**Reply by busslina** [reply]
Class level I mean for example `RearchConsumer.build`, which is a class method. So if you have a `n` class instances you can have `n` different caspules (Similar to dynamic capsules, or inline as you called).

**Reply by GregoryConrad** [reply]
Well yes you can do that via dynamic capsules, but it may cause leaks until I can make those API changes
